modules/javafx.graphics/src/main/java/com/sun/javafx/application/preferences/PreferenceProperties.java

```diff
@@ -226,11 +192,13 @@ public T get() {
         }
 
         /**
-         * Only called by {@link #updateDeferredProperty}, this method doesn't fire a change notification.
+         * Only called from {@link PreferenceProperties#update}, this method doesn't fire a change notification.
          * Change notifications are fired after the new values of all deferred properties have been set.
          */
-        public void setValue(T value) {
-            this.platformValue = value;
+        @SuppressWarnings("unchecked")
+        public void setPlatformValue(Object value) {
```

Comment:
should this method be declared package protected?

Reply:
It doesn't really matter, since it's a nested class. The outer class has access to all members, regardless of visibility. Sometimes I use access modifiers in nested classes to informally document intent (which methods are supposed to be called by the outer class, and which are not). For consistency, I've added the `public` modifier to `fireValueChangeIfNecessary`.

Reply:
you are probably right: it does not matter.

at first I thought that since the instances of `DeferredProperty` are public, one can use reflection to get to its public methods, but as we learned earlier, reflection will likely fail due to this class being unaccessible.